vault/identity_store_oidc.go

```diff
@@ -0,0 +1,743 @@
+			if err != nil {
+				return nil, err
+			}
+			if err := req.Storage.Put(ctx, entry); err != nil {
+				return nil, err
+			}
+		}
+
+		// store named key (which was either just created or updated)
+		entry, err = logical.StorageEntryJSON("oidc-config/named-key/"+name, namedKey)
+		if err != nil {
+			return nil, err
+		}
+		if err := req.Storage.Put(ctx, entry); err != nil {
+			return nil, err
+		}
+
+		return &logical.Response{
+			Data: map[string]interface{}{
+				"rotation_period":  namedKey.RotationPeriod,
+				"verification_ttl": namedKey.Verificationttl,
+				"algorithm":        namedKey.SigningAlgorithm.String(),
+			},
+		}, nil
+	}
+}
+
+// handleOIDCReadKey is used to read an existing key
+func (i *IdentityStore) handleOIDCReadKey() framework.OperationFunc {
+	return func(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+
+		name := d.Get("name").(string)
+
+		entry, err := req.Storage.Get(ctx, "oidc-config/named-key/"+name)
+		if err != nil {
+			return nil, err
+		}
+		if entry != nil {
+			var storedNamedKey NamedKey
+			if err := entry.DecodeJSON(&storedNamedKey); err != nil {
+				return nil, err
+			}
+			return &logical.Response{
+				Data: map[string]interface{}{
+					"rotation_period":  storedNamedKey.RotationPeriod,
+					"verification_ttl": storedNamedKey.Verificationttl,
+					"algorithm":        storedNamedKey.SigningAlgorithm.String(),
+				},
+			}, nil
+		}
+		return logical.ErrorResponse(fmt.Sprintf("no named key was stored at %q", name)), logical.ErrInvalidRequest
+	}
+}
+
+// handleOIDCDeleteKey is used to delete a key
+func (i *IdentityStore) handleOIDCDeleteKey() framework.OperationFunc {
+	return func(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+
+		name := d.Get("name").(string)
+
+		err := req.Storage.Delete(ctx, "oidc-config/named-key/"+name)
+		if err != nil {
+			return nil, err
+		}
+		//todo this is supposed to return  204-no content
+		return nil, nil
+	}
+}
+
+// handleOIDCListKey is used to list named keys
+func (i *IdentityStore) handleOIDCListKeys() framework.OperationFunc {
+	return func(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+		keys, err := req.Storage.List(ctx, "oidc-config/named-key/")
+		if err != nil {
+			return nil, err
+		}
+		return logical.ListResponse(keys), nil
+	}
+}
+
+// handleOIDCRotateKey is used to manually trigger a rotation on the named key
+func (i *IdentityStore) handleOIDCRotateKey() framework.OperationFunc {
+	return func(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+
+		name := d.Get("name").(string)
+		verificationttlInput := d.Get("verification_ttl").(string)
+		var responseVerificationttl string
+
+		// validate inputs and update rotation overrides
+		overrides := make(map[string]interface{})
```

Comment:
I'd avoid this generality until it's deemed necessary. Clearer to just pass in the TTL. A test of >0  in the function is reasonable.